client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -82,56 +81,53 @@ def process(self, instance):
-                    "z": math.radians(rot[2])
-                },
-                "scale": {
-                    "x": scl[0],
-                    "y": scl[1],
-                    "z": scl[2]
-                }
-            }
 
             row_length = 4
             t_matrix_list = cmds.xform(asset, query=True, matrix=True)
-
-            transform_mm = om.MMatrix(t_matrix_list)
-            transform = om.MTransformationMatrix(transform_mm)
-
-            t = transform.translation(om.MSpace.kWorld)
-            t = om.MVector(t.x, t.z, -t.y)
-            transform.setTranslation(t, om.MSpace.kWorld)
-            transform.rotateBy(
-                om.MEulerRotation(math.radians(-90), 0, 0), om.MSpace.kWorld)
-            transform.scaleBy([1.0, 1.0, -1.0], om.MSpace.kObject)
-
-            t_matrix_list = list(transform.asMatrix())
-
+            maya_transform_mm = om.MMatrix(t_matrix_list)
+            convert_transform_mm = om.MMatrix()
+            for i in range(0, row_length):
+                first_row = maya_transform_mm.getElement(i, 0)
+                second_row = maya_transform_mm.getElement(i, 1)
+                third_row = maya_transform_mm.getElement(i, 2)
+                fourth_row = maya_transform_mm.getElement(i, 3)
+                if i == 1:
+                    convert_transform_mm.setElement(i, 0, -first_row)
+                    convert_transform_mm.setElement(i, 1, second_row)
+                    convert_transform_mm.setElement(i, 2, -third_row)
+                    convert_transform_mm.setElement(i, 3, -fourth_row)
+                else:
+                    convert_transform_mm.setElement(i, 0, first_row)
+                    convert_transform_mm.setElement(i, 1, -second_row)
+                    convert_transform_mm.setElement(i, 2, third_row)
+                    convert_transform_mm.setElement(i, 3, fourth_row)
+
+            json_filename = "{}.json".format(instance.name)
+            json_path = os.path.join(stagingdir, json_filename)
             t_matrix = []
-            for i in range(0, len(t_matrix_list), row_length):
-                t_matrix.append(t_matrix_list[i:i + row_length])
+            with lib.maintained_selection():
+                cmds.select(asset, noExpand=True)
+                sel = om.MGlobal.getActiveSelectionList()
+                dagpath = sel.getDependNode(0)
+                ue_transform = om.MFnTransform(dagpath)
+                with lib.maintained_transformation(ue_transform):
+                    # make sure the data doesn't change during context
+                    convert_transform = om.MTransformationMatrix(convert_transform_mm)
+                    final_ue_transform = ue_transform.setTransformation(convert_transform)
+                    t_matrix_list = list(final_ue_transform.transformation().asMatrix())
+
+                    for i in range(0, len(t_matrix_list), row_length):
+                        t_matrix.append(t_matrix_list[i:i + row_length])
```

Comment:
Avoiding the selection may be better?
```suggestion
            sel = om.MSelectionList()
            sel.add(asset)
            dagpath = sel.getDagPath(0)
            ue_transform = om.MFnTransform(dagpath)
            with lib.maintained_transformation(ue_transform):
                # make sure the data doesn't change during context
                convert_transform = om.MTransformationMatrix(convert_transform_mm)
                final_ue_transform = ue_transform.setTransformation(convert_transform)
                t_matrix_list = list(final_ue_transform.transformation().asMatrix())
                for i in range(0, len(t_matrix_list), row_length):
                    t_matrix.append(t_matrix_list[i:i + row_length])
```

I'm not sure what that matrix logic is trying to do... we're using `setTransformation` to set a matrix to then after get that transformation matrix.

Reply:
As far as I can see - this should produce the exact same result:

```suggestion
            convert_transform_mm = list(convert_transform_mm)
            t_matrix_list = []
            for i in range(0, len(t_matrix_list), row_length):
                t_matrix.append(t_matrix_list[i:i + row_length])
```